### PR TITLE
Fix typo in alloc vec comment

### PR DESCRIPTION
**Pull request by pickfire**
cc @the8472

### CODE REVIEW

File: library/alloc/src/vec.rs [hunk 1]
```diff
@@ -2281,7 +2281,7 @@ where
 
         // use try-fold since
         // - it vectorizes better for some iterator adapters
-        // - unlike most internal iteration methods methods it only takes a &mut self
+        // - unlike most internal iteration methods, it only takes a &mut self
         // - it lets us thread the write pointer through its innards and get it back in the end
```
Comment (pickfire) [inline library/alloc/src/vec.rs:2285]: Is this even valid? Looks weird to me.

Comment (the8472) [reply library/alloc/src/vec.rs:2285]: Not sure what in particular you're referring to.

Comment (pickfire) [reply library/alloc/src/vec.rs:2285]: > thread the write pointer through its innards

Comment (the8472) [reply library/alloc/src/vec.rs:2285]: to thread [is a verb](https://en.wiktionary.org/wiki/thread#Verb), and innards is figurative expression for internal components.

**Pull request merged**